docs/src/pages/components/pagination/examples/PaginationHasMorePagesExample.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import { Pagination } from '@aws-amplify/ui-react';
+
+export const PaginationHasMorePagesExample = () => {
+  const [pageTokens, setPageTokens] = React.useState([
+    {
+      res: 'dog',
+      nextToken: 'page2',
+    },
+  ]);
+  const [currPageIndex, setCurrPageIndex] = React.useState(1);
+  const [hasMorePages, setHasMorePages] = React.useState(true);
+
+  const handleNextPage = async () => {
+    if (hasMorePages && currPageIndex === pageTokens.length) {
+      const resultsFromAPI = await mockedAPI();
+
+      if (!resultsFromAPI.nextToken) {
+        setHasMorePages(false);
+      }
+
+      setPageTokens([...pageTokens, resultsFromAPI]);
```

Comment:
Would it be a more accurate real world example of we just put the `resultsFromAPI.nextToken` in this array?
```suggestion
      setPageTokens([...pageTokens, resultsFromAPI.nextToken]);
```